brush-parser/src/parser.rs

```diff
@@ -911,4 +949,29 @@ esac\
 
         Ok(())
     }
+
+    #[test]
+    fn parse_redirection() -> Result<()> {
+        let input = r"echo |& wc";
+
+        let tokens = tokenize_str(input)?;
+        let seq = super::token_parser::pipe_sequence(
+            &Tokens {
+                tokens: tokens.as_slice(),
+            },
+            &ParserOptions::default(),
+            &SourceInfo::default(),
+        )?;
+
+        assert_eq!(seq.len(), 2);
+        assert!(matches!(seq[0], ast::Command::Simple(..)));
+        if let ast::Command::Simple(c) = &seq[0] {
+            let c = c.suffix.as_ref().unwrap();
+            assert!(matches!(
+                c.0[0],
+                ast::CommandPrefixOrSuffixItem::IoRedirect(..)
```

Comment:
Do you think it's worth further asserting that the right Fds are in the `IoRedirect`?